nf_core/lint/files_exist.py

```diff
@@ -242,22 +244,24 @@ def pf(file_path: Union[str, Path]) -> Path:
         else:
             passed.append(f"File not found check: {self._wrap_quotes(file)}")
     # Files that cause an error if they exists together with a certain entry in nextflow.config
-    for file in files_fail_ifinconfig:
-        if str(file[0]) in ignore_files:
+    for file_cond in files_fail_ifinconfig:
+        log.error(f"{self.nf_config=}")
+        if str(file_cond[0]) in ignore_files:
             continue
-        nextflow_config = pf("nextflow.config")
         in_config = False
-        with open(nextflow_config) as f:
-            if file[1] in f.read():
-                in_config = True
-        if pf(file[0]).is_file() and in_config:
-            failed.append(f"File must be removed: {self._wrap_quotes(file[0])}")
-        elif pf(file[0]).is_file() and not in_config:
-            passed.append(f"File found check: {self._wrap_quotes(file[0])}")
-        elif not pf(file[0]).is_file() and not in_config:
-            failed.append(f"File not found check: {self._wrap_quotes(file[0])}")
-        elif not pf(file[0]).is_file() and in_config:
-            passed.append(f"File not found check: {self._wrap_quotes(file[0])}")
+        config_key, config_value = list(file_cond[1].items())[0]
+        if config_key in self.nf_config and config_value in self.nf_config[config_key]:
+            in_config = True
+            log.debug(f"Found {config_key} in nextflow.config with value {config_value}")
+            in_config = True
+        if pf(file_cond[0]).is_file() and in_config:
+            failed.append(f"File must be removed: {self._wrap_quotes(file_cond[0])}")
+        elif pf(file_cond[0]).is_file() and not in_config:
+            passed.append(f"File found check: {self._wrap_quotes(file_cond[0])}")
+        elif not pf(file_cond[0]).is_file() and not in_config:
+            failed.append(f"File not found check: {self._wrap_quotes(file_cond[0])}")
+        elif not pf(file_cond[0]).is_file() and in_config:
+            passed.append(f"File not found check: {self._wrap_quotes(file_cond[0])}")
     # Files that cause a warning if they exist
     for file in files_warn_ifexists:
         if file in ignore_files:
```

Comment:
```suggestion
        if str(file) in ignore_files:
```

Reply:
ignore_files are already be strings I think

Reply:
The files in `files_warn_ifexists` are paths, so I think we have to convert them to strings to compare with the ones in `ignore_files`